firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/FirebaseAppDistribution.java

```diff
@@ -62,18 +115,129 @@ public static FirebaseAppDistribution getInstance() {
    * starts an installation If the latest release is an AAB, directs the tester to the Play app to
    * complete the download and installation.
    *
-   * @throws an {@link Status.UPDATE_NOT_AVAILABLE_ERROR} exception if no new release is cached from
-   *     checkForUpdate
+   * @throws FirebaseAppDistributionException with UPDATE_NOT_AVAIALBLE exception if no new release
+   *     is cached from checkForUpdate
    */
   @NonNull
   public UpdateTask updateApp() {
     return (UpdateTask) Tasks.forResult(new UpdateState(0, 0, UpdateStatus.PENDING));
   }
 
+  private boolean supportsCustomTabs(Context context) {
+    Intent customTabIntent = new Intent("android.support.customtabs.action.CustomTabsService");
+    customTabIntent.setPackage("com.android.chrome");
+    List<ResolveInfo> resolveInfos =
+        context.getPackageManager().queryIntentServices(customTabIntent, 0);
+    return resolveInfos != null && !resolveInfos.isEmpty();
+  }
+
+  public static @NonNull String getApplicationName(@NonNull Context context) {
+    try {
+      return context.getApplicationInfo().loadLabel(context.getPackageManager()).toString();
+    } catch (Exception e) {
+      Log.e(TAG, "Unable to retrieve App name");
+      return "";
+    }
+  }
+
+  private void openSignIn(Uri uri) {
+    currentlySigningIn = true;
+    if (supportsCustomTabs(firebaseApp.getApplicationContext())) {
+      // If we can launch a chrome view, try that.
+      CustomTabsIntent.Builder builder = new CustomTabsIntent.Builder();
+
```

Comment:
Out of curiosity, do we need this animation? How does it behave if we don't set any animation?

Its not a super big deal but unless we really need it, we should not have it. Animations can be a bit wonky on older devices.

Reply:
Without the animation the transition looks like how it does for a normal browser where it looks like it's popping up a new app. It's not needed so I'll remove them